trl/trainer/ppo_trainer.py

```diff
@@ -1094,6 +1096,16 @@ def record_step_stats(self, kl_coef: float, **data):
             "ppo/std_scores": std_scores,
         }
 
+        if self.config.log_token_distribution:
+            # Log text properties
+            query_lens = torch.tensor([len(query) for query in data["queries"]], dtype=torch.float)
+            response_lens = torch.tensor([len(response) for response in data["responses"]], dtype=torch.float)
+
+            stats["tokens/query_len_mean"] = torch.mean(query_lens).cpu().numpy().item()
+            stats["tokens/query_len_std"] = torch.std(query_lens).cpu().numpy().item()
+            stats["tokens/responses_len_mean"] = torch.mean(response_lens).cpu().numpy().item()
+            stats["tokens/responses_len_std"] = torch.std(response_lens).cpu().numpy().item()
```

Comment:
similar to rewards i would also log the distribution so we can look at histograms in the logs.